src/api/sso/src/middleware.js

```diff
@@ -104,16 +104,21 @@ module.exports.createTelescopeUser = function createTelescopeUser() {
       { returning: 'minimal' }
     );
 
-    if (userError) {
-      logger.warn(
-        { code: userError.code, details: userError.details, id, data: req.body },
-        userError.message
-      );
-      next(createError(400, 'unable to create user'));
+    if (response.error) {
+      const {
+        status,
+        error: { details, message },
+      } = response;
+      logger.warn({ status, details, id, data: req.body }, message);
+      if (status === 409) {
+        next(createError(status, 'Display name or GitHub account has already been registered'));
+      } else {
+        next(createError(status, 'Unable to create a Telescope profile'));
+      }
       return;
     }
 
-    const { error: feedError } = await supabase.from('feeds').insert(
+    response = await supabase.from('feeds').insert(
```

Comment:
Instead of re-using `response` here, I wonder if you should break these two sections into separate functions that you call together.

Reply:
Ok, let me try that, the test is failing btw.